lib/AST/Identifier.cpp

```diff
@@ -28,8 +28,9 @@ raw_ostream &llvm::operator<<(raw_ostream &OS, Identifier I) {
 }
 
 raw_ostream &llvm::operator<<(raw_ostream &OS, DeclBaseName I) {
-  // TODO: Handle special names
-  return OS << I.getIdentifier();
+  if (I.empty())
+    return OS << "_";
```

Comment:
Yeah, makes sense. `userFacingStr` now returns "_" for the empty name since we require "_" for `<<` for quite a few diagnostics IIRC